### PR TITLE
Update the WASM compatibility tests for `0.36` release

### LINKED ISSUE #2198: Add forward compatibility test for `fuel-core 0.36.0`
After release of `fuel-core 0.36.0` we need to add forward compatibility test for it.

image

### PR DESCRIPTION
**Pull request by rafal-ch**
Closes #2198.

## Description
This test upgrades the forward compatibility test by replacing the "genesis" nodes (which used v0.26) with the nodes that use v36 of the client.

It also adds one backward compatibility test: `latest_binary_is_backward_compatible_and_follows_blocks_created_by_v36_binary()`

### Before requesting review
- [X] I have reviewed the code myself

### CODE REVIEW

File: version-compatibility/forkless-upgrade/src/backward_compatibility.rs [hunk 1]
```diff
@@ -47,15 +47,14 @@ async fn latest_binary_is_backward_compatible_and_can_load_testnet_config() {
 }
 
 #[tokio::test]
-async fn latest_binary_is_backward_compatible_and_follows_blocks_created_by_genesis_binary(
-) {
+async fn latest_binary_is_backward_compatible_and_follows_blocks_created_by_v36_binary() {
```
Comment (xgreenx) [inline version-compatibility/forkless-upgrade/src/backward_compatibility.rs:50]: It still would be nice to have one more backward compatibility tests that produces blocks by `fuel-core 0.26.0`

Comment (rafal-ch) [reply version-compatibility/forkless-upgrade/src/backward_compatibility.rs:121]: Genesis test is restored in https://github.com/FuelLabs/fuel-core/pull/2271/commits/13f26532454896633cac084c936d265c100049fc

**Review by rymnc — APPROVED** [review]

File: version-compatibility/forkless-upgrade/src/backward_compatibility.rs [hunk 2]
```diff
@@ -109,7 +112,76 @@ async fn latest_binary_is_backward_compatible_and_follows_blocks_created_by_gene
...
+    let v36_node = Version36FuelCoreDriver::spawn(&[
+        "--service-name",
+        "V36Producer",
+        "--debug",
+        "--poa-instant",
+        "true",
+        "--consensus-key",
+        POA_SECRET_KEY,
+        "--snapshot",
+        IGNITION_TESTNET_SNAPSHOT,
```
Comment (xgreenx) [inline version-compatibility/forkless-upgrade/src/backward_compatibility.rs:137]: This snapshot using the state transition function from `fuel-core 0.26.0`. Maybe we want to add a new snapshot for `fuel-core 0.36`?=)

**Review by xgreenx — APPROVED** [review]

**Review by AurelienFT — APPROVED** [review]
LGTM

**Pull request merged**